src/basic/model_configuration.f90

```diff
@@ -738,13 +754,32 @@ MODULE model_configuration
 
     ! Grounding line treatment
     LOGICAL             :: do_subgrid_BMB_at_grounding_line_config      = .FALSE.                          ! Whether or not to apply basal melt rates under a partially floating grounding line
+    CHARACTER(LEN=256)  :: choice_BMB_subgrid_config                    = ''                               ! Choice of sub-grid BMB scheme: "FCMP", "PMP", "NMP" (following Leguy et al., 2021)
 
     ! Choice of BMB model
     CHARACTER(LEN=256)  :: choice_BMB_model_NAM_config                  = 'uniform'
     CHARACTER(LEN=256)  :: choice_BMB_model_EAS_config                  = 'uniform'
     CHARACTER(LEN=256)  :: choice_BMB_model_GRL_config                  = 'uniform'
     CHARACTER(LEN=256)  :: choice_BMB_model_ANT_config                  = 'uniform'
 
+    ! Prescribed BMB forcing
+    CHARACTER(LEN=256)  :: choice_BMB_prescribed_NAM_config             = ''
+    CHARACTER(LEN=256)  :: choice_BMB_prescribed_EAS_config             = ''
+    CHARACTER(LEN=256)  :: choice_BMB_prescribed_GRL_config             = ''
+    CHARACTER(LEN=256)  :: choice_BMB_prescribed_ANT_config             = ''
+
+    ! Files containing prescribed BMB forcing
+    CHARACTER(LEN=256)  :: filename_BMB_prescribed_NAM_config           = ''
+    CHARACTER(LEN=256)  :: filename_BMB_prescribed_EAS_config           = ''
+    CHARACTER(LEN=256)  :: filename_BMB_prescribed_GRL_config           = ''
+    CHARACTER(LEN=256)  :: filename_BMB_prescribed_ANT_config           = ''
+
+    ! Timeframes for reading prescribed BMB forcing from file (set to 1E9_dp if the file has no time dimension)
```

Comment:
I'll see if Fortran's namelist functionality can read NaN, would be a lot cleaner.